types/metro-resolver/index.d.ts

```diff
@@ -3,9 +3,32 @@
 // Definitions by: Adam Foxman <https://github.com/afoxman>
 //                 Tommy Nguyen <https://github.com/tido64>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
 
-export * from './types';
+export {
+    AssetFileResolution,
+    CustomResolutionContext,
+    CustomResolver,
+    CustomResolverOptions,
+    DoesFileExist,
+    FileAndDirCandidates,
+    FileCandidates,
+    FileResolution,
+    IsAssetFile,
+    ResolutionContext,
+    Resolution,
+    ResolveAsset,
+    Result,
 
-import { ResolutionContext, Resolution } from './types';
+    // didn't exported in the origin code
+    FileContext,
+    FileOrDirContext,
+    HasteContext,
+    ModulePathContext,
```

Comment:
Should we be exporting these if they were not exported upstream?

Reply:
Hummmm...It is exported previously, in order to keep the compatibility, I think I need to keep it